redbot/core/commands/converter.py

```diff
@@ -162,140 +185,262 @@ class APIToken(discord.ext.commands.Converter):
-    The units should be in order from largest to smallest.
-    This works with or without whitespace.
+    def get_dict_converter(*expected_keys: str, delims: Optional[List[str]] = None) -> Type[dict]:
+        """
+        Returns a typechecking safe `DictConverter` suitable for use with discord.py
+        """
 
-    See `parse_timedelta` for more information about how this functions.
+        class PartialMeta(type):
+            __call__ = functools.partialmethod(
+                type(DictConverter).__call__, *expected_keys, delims=delims
+            )
 
-    Attributes
-    ----------
-    maximum : Optional[timedelta]
-        If provided, any parsed value higher than this will raise an exception
-    minimum : Optional[timedelta]
-        If provided, any parsed value lower than this will raise an exception
-    allowed_units : Optional[List[str]]
-        If provided, you can constrain a user to expressing the amount of time
-        in specific units. The units you can chose to provide are the same as the
-        parser understands: `weeks` `days` `hours` `minutes` `seconds`
-    default_unit : Optional[str]
-        If provided, it will additionally try to match integer-only input into
-        a timedelta, using the unit specified. Same units as in `allowed_units`
-        apply.
-    """
+        class ValidatedConverter(DictConverter, metaclass=PartialMeta):
+            pass
 
-    def __init__(self, *, minimum=None, maximum=None, allowed_units=None, default_unit=None):
-        self.allowed_units = allowed_units
-        self.default_unit = default_unit
-        self.minimum = minimum
-        self.maximum = maximum
+        return ValidatedConverter
 
-    async def convert(self, ctx: "Context", argument: str) -> timedelta:
-        if self.default_unit and argument.isdecimal():
-            delta = timedelta(**{self.default_unit: int(argument)})
-        else:
-            delta = parse_timedelta(
-                argument,
-                minimum=self.minimum,
-                maximum=self.maximum,
-                allowed_units=self.allowed_units,
+
+if TYPE_CHECKING:
+    TimedeltaConverter = timedelta
+else:
+
+    class TimedeltaConverter(dpy_commands.Converter):
+        """
+        This is a converter for timedeltas.
+        The units should be in order from largest to smallest.
+        This works with or without whitespace.
+
+        See `parse_timedelta` for more information about how this functions.
+
+        Attributes
+        ----------
+        maximum : Optional[timedelta]
+            If provided, any parsed value higher than this will raise an exception
+        minimum : Optional[timedelta]
+            If provided, any parsed value lower than this will raise an exception
+        allowed_units : Optional[List[str]]
+            If provided, you can constrain a user to expressing the amount of time
+            in specific units. The units you can chose to provide are the same as the
```

Comment:
Not part of PR but since I noticed it, figured I can mention it 🤷‍♂ 
```suggestion
            in specific units. The units you can choose to provide are the same as the
```